docs/customruns.md

```diff
@@ -0,0 +1,413 @@
+
+- [Overview](#overview)
+- [Configuring a `CustomRun`](#configuring-a-customrun)
+  - [Specifying the target Custom Task](#specifying-the-target-custom-task)
+  - [Cancellation](#cancellation)
+  - [Specifying `Timeout`](#specifying-timeout)
+  - [Specifying `Retries`](#specifying-retries)
+  - [Specifying Parameters](#specifying-parameters)
+  - [Specifying Workspaces](#specifying-workspaces)
+  - [Specifying Service Account](#specifying-a-serviceaccount)
+- [Monitoring execution status](#monitoring-execution-status)
+  - [Monitoring `Results`](#monitoring-results)
+- [Code examples](#code-examples)
+  - [Example `CustomRun` with a referenced custom task](#example-customrun-with-a-referenced-custom-task)
+  - [Example `CustomRun` with an unnamed custom task](#example-customrun-with-an-unnamed-custom-task)
+  - [Example of specifying parameters](#example-of-specifying-parameters)
+
+# Overview
+
+A `CustomRun` allows you to instantiate and execute a [Custom
+Task](https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md),
+which can be implemented by a custom task controller running on-cluster. Custom
+Tasks can implement behavior that doesn't correspond directly to running a
+workload in a `Pod` on the cluster. For Pod-based on-cluster workloads, you
+should use a [`TaskRun`](taskruns.md).
+
+In order for a `CustomRun` to actually execute, there must be a custom task
+controller running on the cluster that is responsible for watching and updating
+`CustomRun`s which reference their type. If no such controller is running, `CustomRun`s
+will have no `.status` value and no further action will be taken.
+
+## Configuring a `CustomRun`
+
+A `CustomRun` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`][kubernetes-overview] - Specifies the API version, for example
+    `tekton.dev/v1beta1`.
+  - [`kind`][kubernetes-overview] - Identifies this resource object as a `CustomRun` object.
+  - [`metadata`][kubernetes-overview] - Specifies the metadata that uniquely identifies the
+    `CustomRun`, such as a `name`.
+  - [`spec`][kubernetes-overview] - Specifies the configuration for the `CustomRun`.
+    - [`customRef`](#1-specifying-the-target-custom-task-with-customref) - Specifies the type and
+      (optionally) name of the custom task type to execute.
+    - [`customSpec`](#2-specifying-the-target-custom-task-by-embedding-its-spec) - Embed the custom task resource spec
+      directly in a `CustomRun`.
+- Optional:
+  - [`timeout`](#specifying-timeout) - specifies the maximum duration of a single execution
+    of a `CustomRun`.
+  - [`retries`](#specifying-retries) - specifies the number of retures upon `CustomRun`failure.
+  - [`params`](#specifying-parameters) - Specifies the desired execution
+    parameters for the custom task.
+  - [`serviceAccountName`](#specifying-a-serviceaccount) - Specifies a `ServiceAccount`
+    object that provides custom credentials for executing the `CustomRun`.
```

Comment:
I'd suggest removing "provides custom credentials" since service accounts get permissions to do things via RoleBindings and ClusterRoleBindings or workload identity.